### DISCUSSION TITLE
Laravel Livewire 3 | Extracting a form object | Array Validation | No property found for validation: [photos.*]

### DISCUSSION THREAD

**Discussion by madushancs** [livewire/livewire#6489] [1. Help]
Hello All,

How can I validate Array when using the livewire form?

`<input" accept="image/png, image/jpeg" type="file" wire:model="UserRegisterForm.photos"  multiple />`

`<?php

namespace App\Livewire;

use App\Livewire\Forms\UserRegisterForm;
use Livewire\Attributes\Rule;
use Livewire\Attributes\Title;
use Livewire\Component;
use Livewire\WithFileUploads;

class Register extends Component
{
    use WithFileUploads;
    public UserRegisterForm $UserRegisterForm; 

    public function save()
    {
        $this->UserRegisterForm->validate();
        $this->UserRegisterForm->store(); 
    }

    public function render()
    {
        return view('livewire.register');
    }
}`

`namespace App\Livewire\Forms;
use Livewire\Attributes\Rule;
use Livewire\Form;
use Illuminate\Support\Str;
use Livewire\WithFileUploads;

class UserRegisterForm extends Form
{
    use WithFileUploads; 

    #[Rule('nullable|sometimes|image|max:1024', message: 'Please attempt with an image smaller than 1MB.')]
    public $photo; 

    #[Rule(['photos.*' => 'nullable|sometimes|image|max:10'])]
    public $photos = [];

    public $FamilyPhotoId = '';

    public function store()
    { 
        if ($this->photo) {
            $filePath = $this->photo->store('individual', 'public');
        }

        if ($this->photos) {
            foreach ($this->photos as $poto) {
                $potoPath = $poto->store('family', 'public'); 
            }
        } 
    }
}`

When I add a image for $photos I get a error as below image,

![image](https://github.com/livewire/livewire/assets/45415384/559946de-0912-4fa9-9e2d-92332a8816d0)

my packages:
        "php": "^8.1",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^10.10",
        "laravel/sanctum": "^3.2",
        "laravel/tinker": "^2.8",
        "livewire/livewire": "3.0.0-beta.11"

**Comment by jleonardolemos** [comment]
Same error for array validation.
I have tested the documentation example from inside a form Class:

```
use Livewire\Attributes\Rule;
 
#[Rule([
    'titles' => 'required',
    'titles.*' => 'required|min:5',
], message: [
    'required' => 'The :attribute is missing.',
    'titles.required' => 'The :attribute are missing.',
    'min' => 'The :attribute is too short.',
], attribute: [
    'titles.*' => 'title',
])]
public $titles = [];
```

The same validation showed in the docs.

**Comment by cybertcode** [comment]
Estoy siguiendo la documentación de livewire 3 y tengo el mismo problema : No property found for validation: [images.*]
    #[Rule(['images.*' => 'image|max:1024'])]
    public $images = [];
_________________________________________________________________
            <input wire:model='form.images'
                class="block w-full text-sm text-gray-900 border border-gray-300 rounded-lg cursor-pointer bg-gray-50 dark:text-gray-400 focus:outline-none dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400"
                aria-describedby="images_help" id="images" type="file" multiple>
            <label class="mt-1 text-sm text-gray-500 dark:text-gray-300">subir en .JPG ó .PNG</label>
            @error('form.images.*')
            <p class="mt-2 text-sm text-red-600 dark:text-red-500"><span
                        class="font-medium">Oops!</span>{{ $message }}</p>
            @enderror
------------------------------------------------------------------------------------------

**Comment by michaelkonstantinou** [comment]
I think the documentation is either wrong or unclear. I had the same problem, here is how I fixed it:

If you have this in the blade.php file
`<input wire:module.live="names" />`

Then the rule in the form component must be as follows:
```php
#[Rule([
        'form.names' => 'required',
        'form.names.*' => [
            'required',
            'min:3'
        ],
    ])]
```

Notice that, you need to specify the the property name is `form.[NAME_HERE]` and not just the name of the property (emphasis that you need to add **form** in front of the property's name.) 

Hence in your situation, since you have the variable `$images`, you should add the following rules

```php
#[Rule([
        'form.images' => 'required',
        'form.images.*' => [image|max:1024],
    ])]
```

**Reply by GregOstry** [reply]
Thanks. That's a really good tip! It's definitely not clear when you read the documentation about the form objects.
